tests/Providers/AbstractProviderTest.php

```diff
@@ -18,15 +18,15 @@ class AbstractProviderTest extends TestCase
     private $client;
 
 
-    public function setUp(): void
+    protected function setUp(): void
```

Comment:
Why does it matter? What is improved by these not being visible outside of the test class?

Reply:
I think it's that reason and it should only let this method be manipulated by extending this class.

Reply:
So there's no benefit?

Reply:
The benefit is about letting developers prevent access this method directly.

It should inherit current class to access that.

Reply:
Sorry I'm struggling to understand in what scenario would a developer create an instance of a test class?
Additionally, if they are doing that, why would it be useful to prevent them from calling `setUp()`?

Reply:
It should inherit the `PHPUnit\Framework\TestCase` class to access/override the `setUp` method.

Some developers need to override the `TestCase::setUp` method and they will inherit that class to complete that.

Using the `protected` for `setUp` method to let developers know that it should use inheriting approach to override `setUp` method and also follow the [official PHPUnit doc](https://phpunit.readthedocs.io/en/8.5/fixtures.html#more-setup-than-teardown) introduction about `setUp` and `tearDown`.